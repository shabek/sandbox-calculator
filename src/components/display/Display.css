.display {
  background-color: #fff;
  border: 1px solid #d3d3d3;
  border-radius: 0.5rem;
  color: #000;
  font-size: 2rem;
  font-weight: 400;
  letter-spacing: 0.015em;
  line-height: 1;
  overflow: hidden;
  padding: 1rem;
  text-align: right;
  width: 100%;
}
