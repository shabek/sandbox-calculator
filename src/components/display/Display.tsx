import { styled } from "@mui/material";

/**
 * Display provided value in decorated container.
 */

const DisplayFrame = styled("div")(({ theme }) => ({
  fontSize: "3em",
  height: "2em",
  overflow: "hidden",
  padding: theme.spacing(2),
  textAlign: "right",
  width: "100%",
}));

type DisplayProps = {
  displayed: string;
};

function Display({ displayed }: DisplayProps) {
  return <DisplayFrame title="Display">{displayed}</DisplayFrame>;
}

export default Display;
