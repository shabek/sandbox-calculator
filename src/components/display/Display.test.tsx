import { render, screen } from "@testing-library/react";
import Display from "./Display";

describe("<Display />", () => {
  test("renders", () => {
    const { asFragment } = render(<Display displayed="Lorem Ipsum" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays provided text", () => {
    render(<Display displayed="Lorem Ipsum" />);
    expect(screen.getByLabelText("Display")).toHaveTextContent("Lorem Ipsum");
  });

  test("creates correct aria attributes", () => {
    render(<Display displayed="Lorem Ipsum" />);

    expect(screen.getByLabelText("Display")).toHaveAttribute(
      "aria-live",
      "polite"
    );
  });
});
