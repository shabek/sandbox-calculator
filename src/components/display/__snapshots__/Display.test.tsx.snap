// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Display /> > renders 1`] = `
<DocumentFragment>
  <section
    aria-label="Display"
    aria-live="polite"
    class="display"
  >
    Lorem Ipsum
  </section>
</DocumentFragment>
`;
